docs/external.rst

```diff
@@ -3,66 +3,147 @@
 Interacting with external arrays
 ================================
 
-**External arrays** refer to ``numpy.ndarray`` or ``torch.Tensor``.
+Introduction
+------------
 
-Conversion between Taichi fields and external arrays
-----------------------------------------------------
+Although Taichi fields are mainly used in Taichi-scope, in some cases efficiently manipulating Taichi field data in Python-scope could also be helpful.
 
-Use ``to_numpy``/``from_numpy``/``to_torch``/``from_torch``:
+So we provide a way to export and import data in Taichi fields via NumPy arrays.
+This allows you to interact with the data in Python-scope with other packages
+that supports NumPy.
```

Comment:
This section is the introduction for interacting with the whole **external array**, so it does not sound right to directly talk about **providing interfaces for NumPy**, as NumPy is only part of **external array**.

What about inferring that Numpy is the most common cases, and introduce the 2 code blocks as the example.

```suggestion
We provide various interfaces to copy the data between Taichi fields and external arrays.
The most typical case maybe copying between Tachi fields and Numpy arrays. So, let's take a look at the 2 examples below.
```